### PR TITLE
Issue 14 embed

### LINKED ISSUE #14: [Block] - Embed
- [ ] Mobile version
- [ ] Desktop version
- [ ] Autoblock it in a way that if the author just puts in a link of brightcove video or Flippingbook url in the word document, autobuild the block.

**Note**: if it makes it more intuitive to split these 2 into 2 separate blocks (video, embed) that should be OK too. Also, check with customer incase we have more variations of embeds that we want to implement before hand.

<hr>

**Variations**
1. **Brightcove Video**
See sample at crn.com

Screenshot 2023-09-14 at 10 04 03 AM

2. **Flipping Book**
See sample at https://www.crn.com/crnmagazine/index.htm

Screenshot 2023-09-14 at 10 05 30 AM

3.**Issuu**
See sample at https://www.crn.com/crnmagazine/index.htm

Screenshot 2023-09-20 at 2 56 56 PM

### PR DESCRIPTION
**Pull request by justinli-ensemble**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #14

Test URLs:
- Before: https://main--channelco-crn-com--hlxsites.hlx.page/drafts/kunwar/embed
- After: https://issue-14-embed--channelco-crn-com--hlxsites.hlx.page/
-  After: https://issue-14-embed--channelco-crn-com--hlxsites.hlx.page/drafts/kunwar/embed

### CODE REVIEW

**Comment by kunwarsaluja** [comment]
@mfrisbey @varunmitra Can i get another set of eyes on this PR by @justinli-ensemble  ? I don't understand why the TBT is fluctuating alot here. Can you suggest what can we do here to get this TBT lower ?

**Comment by kunwarsaluja** [comment]
Identified a couple global layout problems and now raised a bug #60 for @samuelopra to fix which I think is contributing to this issue too (Around CLS atleast) , #60

File: blocks/brightcove/brightcove.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      <video-js
+        data-account="${brightcovedataaccount}"
+        data-player="${player}"
+        controls=""
+        data-embed="default"
+        data-video-id="${video || ''}"
+        data-playlist-id="${playlist || ''}"
+        data-application-id=""
+        class="vjs-fluid"></video-js>
+      <script src="//players.brightcove.net/${brightcovedataaccount}/${player}_default/index.min.js"></script>
```
Comment (mfrisbey) [inline blocks/brightcove/brightcove.js:19]: Could embedding this script in the block be causing problems? I believe this happens during the eager phase?

**Review by mfrisbey — COMMENTED** [review]
I ran before/after tests on the URL `/drafts/kunwar/embed`, and it's definitely something in this PR that's causing slowness.

Seems like script evaluation and third party code is causing problems:

Screenshot 2023-09-26 at 11 40 16 AM

Screenshot 2023-09-26 at 11 40 31 AM

I don't see any explicit script inclusions for cloudfront. We could systematically remove the `iframe` and the `script` tag that I pointed out to see if removing one or the other fixes it; that way we at least know where to start looking.

**Comment by kunwarsaluja** [comment]
@mfrisbey @justinli-ensemble : Now has moved the implementation to use  intersectionObserver as the brightcove video is usually outside the viewpoint so we can delay the load time of script which is taking 10+seconds to fully load hence increasing the TBT.

**Pull request closed without merging**